File: backend/src/main/java/com/example/backend/models/ChatMessage.java
package com.example.backend.models;

import org.springframework.data.annotation.Id;

import java.time.LocalDateTime;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;

import lombok.Data;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
@Data
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String message;
    private User sender;
    private LocalDateTime sendingTime;

}

File: app/src/main/java/com/example/dropletbarterapp/models/ChatMessage.kt
package com.example.dropletbarterapp.models

import lombok.Data
import lombok.Getter
import lombok.RequiredArgsConstructor
import lombok.Setter
import org.springframework.data.annotation.Id
import java.time.LocalDateTime
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType

class ChatMessage(
    val message: String,
    val sender: User,
    val sendingTime: LocalDateTime,
    val read: Boolean = false
)